support/cas-server-support-spnego/src/main/java/org/apereo/cas/config/SpnegoConfiguration.java

```diff
@@ -65,40 +64,57 @@ public BeanContainer<Authentication> spnegoAuthentications(
+                    jcifsSettings.setJcifsDomain(p.getJcifsDomain());
+                    jcifsSettings.setJcifsDomainController(p.getJcifsDomainController());
+                    jcifsSettings.setJcifsNetbiosCachePolicy(p.getCachePolicy());
+                    jcifsSettings.setJcifsNetbiosWins(p.getJcifsNetbiosWins());
+                    jcifsSettings.setJcifsPassword(p.getJcifsPassword());
+                    jcifsSettings.setJcifsServicePassword(p.getJcifsServicePassword());
+                    jcifsSettings.setJcifsServicePrincipal(p.getJcifsServicePrincipal());
+                    jcifsSettings.setJcifsSocketTimeout(Beans.newDuration(p.getTimeout()).toMillis());
+                    jcifsSettings.setJcifsUsername(p.getJcifsUsername());
+                    return new Authentication(jcifsSettings.getProperties());
+                })
+                .collect(Collectors.toList());
+    }
+
+    @RefreshScope(proxyMode = ScopedProxyMode.DEFAULT)
+    @Bean
+    @ConditionalOnMissingBean(name = "spnegoAuthenticationsPool")
+    public BlockingQueue<List<Authentication>> spnegoAuthenticationsPool(
+            final CasConfigurationProperties casProperties,
+            final ConfigurableApplicationContext applicationContext) {
+        val spnegoProperties = casProperties.getAuthn().getSpnego();
+        val poolSize = spnegoProperties.getPoolSize();
+        val spnegoAuthenticationPool = new ArrayBlockingQueue<List<Authentication>>(poolSize);
+        for (int i = 0; i < poolSize; i++) {
+            val list = buildSpnegoAuthentications(casProperties, applicationContext);
+            spnegoAuthenticationPool.add(list);
+        }
+        return spnegoAuthenticationPool;
     }
 
     @Bean
     @RefreshScope(proxyMode = ScopedProxyMode.DEFAULT)
     @ConditionalOnMissingBean(name = "spnegoHandler")
     public AuthenticationHandler spnegoHandler(
-        @Qualifier("spnegoAuthentications")
-        final BeanContainer<Authentication> spnegoAuthentications,
+        @Qualifier("spnegoAuthenticationsPool")
+        final BlockingQueue<List<Authentication>> spnegoAuthenticationsPool,
         @Qualifier("spnegoPrincipalFactory")
         final PrincipalFactory spnegoPrincipalFactory,
         @Qualifier(ServicesManager.BEAN_NAME)
         final ServicesManager servicesManager,
         final CasConfigurationProperties casProperties) {
         val spnegoProperties = casProperties.getAuthn().getSpnego();
+        val poolTimeout = Beans.newDuration(spnegoProperties.getPoolTimeout()).toMillis();
         return new JcifsSpnegoAuthenticationHandler(spnegoProperties.getName(),
-            servicesManager, spnegoPrincipalFactory, spnegoAuthentications.toList(),
+            servicesManager, spnegoPrincipalFactory, spnegoAuthenticationsPool,
             spnegoProperties.isPrincipalWithDomainName(), spnegoProperties.isNtlmAllowed(),
-            spnegoProperties.getOrder());
+            spnegoProperties.getOrder(), poolTimeout);
```

Comment:
Also, it would be better if you rework the handler and just pass it `spnegoProperties` altogether.